server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorEncoderDecoder.java

```diff
@@ -70,7 +67,7 @@ public static void decodeDenseVector(BytesRef vectorBR, float[] vector) {
         }
         ByteBuffer byteBuffer = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length);
         for (int dim = 0; dim < vector.length; dim++) {
-            vector[dim] = byteBuffer.getFloat();
+            vector[dim] = byteBuffer.getFloat((dim * Float.BYTES) + vectorBR.offset);
```

Comment:
The vectorBR.offset is already taken into account to create the ByteBuffer, so we shouldn't add it there again?

Also unrelated to your PR, I would expect `ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length).asFloatBuffer().get(vector)` to run faster than reading a float at a time.

Reply:
@jpountz I just asked @mayya-sharipova about this same thing. Using `getFloat(int)` ignores the current position of the buffer. It may make sense to wrap without the offset and length parameters.

Reply:
> Also unrelated to your PR, I would expect ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length).asFloatBuffer().get(vector) to run faster than reading a float at a time.

@jpountz it doesn't. This is 30% faster than making it a float buffer. For some reason, getting the absolute is faster. @ChrisHegarty might have an intuition as to why.

Reply:
Yes, it's the endianness of the bytebuffers we're using here - big in this case. It's most efficient to use the native endianness, otherwise byte swapping will occur.   Since these are already in the index, is it possible to switch them to little?

Reply:
@jpountz here are some benchmark results:

So, average time to calculate latency for decoding `float[]` from ByteBuffers. Here is the benchmark code (if you can spot any wonkiness): https://gist.github.com/benwtrent/29cc0338cd851c345cace5c486095507

Direct Read (via index in the buffer) is always faster. The floatbuffer vs. byte buffer are almost identical for both decoding kinds (iteration vs. direct read).

```
Benchmark                                                          (floatArraySizes)  Mode  Cnt         Score         Error  Units
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferDirectRead                  96  avgt    5   3348858.476 ±   13314.154  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferDirectRead                 768  avgt    5  19960837.116 ±  137071.324  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferDirectRead                2048  avgt    5  49142580.882 ±  216614.762  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferIteration                   96  avgt    5   4992161.167 ±   47263.300  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferIteration                  768  avgt    5  33762525.212 ±  110609.781  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeByteBufferIteration                 2048  avgt    5  89409177.001 ± 1122561.441  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferDirectRead                 96  avgt    5   3529213.563 ±   24635.601  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferDirectRead                768  avgt    5  18152123.115 ±   49712.077  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferDirectRead               2048  avgt    5  46610729.883 ± 2462771.340  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferIteration                  96  avgt    5   5085876.957 ±   75157.360  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferIteration                 768  avgt    5  32359519.955 ±  399297.947  ns/op
ByteBufferFloatDecodeLatencyBenchmark.decodeFloatBufferIteration                2048  avgt    5  86083215.393 ± 1120197.606  ns/op
```